Sat_Training.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[47]:
+
+
+import pandas as pd
+import numpy as np
+from scipy.stats import ttest_rel
+from scipy.stats import ttest_ind
+
+
+# In[5]:
+
+
+df=pd.read_csv('C:/Users/skhom4875/Desktop/Training/satday-4-sept-quiz/bike_sharing_hours.csv')
+
+
+# In[6]:
+
+
+df.head()
+
+
+# In[8]:
+
+
+df.shape
+
+
+# In[54]:
+
+
+df.mnth.value_counts()
+
+
+# In[10]:
+
+
+df.info()
+
+
+# In[12]:
+
+
+df['dteday']=pd.to_datetime(df['dteday'])
+
+
+# In[13]:
+
+
+df['dteday'].describe()
+
+
+# In[29]:
+
+
+import datetime as dt
+
+
+# In[55]:
+
+
+#extracting cnt for first 6 months
+ls=[]
+for i in range(len(df)):
+    
+    if df.dteday[i].month in [1,2,3,4,5,6] and  (df.dteday[i].day < 8):  
+        ls.append(df['cnt'][i])
+
+
+# In[59]:
+
+
+#extracting cnt for last 6 months
+ls2=[]
+for i in range(len(df)):
+    
+    if df.dteday[i].month in [7,8,9,10,11,12] and  (df.dteday[i].day < 8):  
+        ls2.append(df['cnt'][i])
+
+
+# In[61]:
+
+
+len(ls)
+
+
+# In[62]:
+
+
+len(ls2)
+
+
+# In[67]:
+
+
+from scipy.stats import levene
+levene(ls,ls2)
+
+
+# We  reject the null hypothesis. Variance is not equal
+
+# In[83]:
+
+
+#We will perform unpaired t test
+ttest_ind(ls,ls2,equal_var=False)
+
+
+# There is significant difference between average number of rented bikes in first 7 days of months for first 6 months Vs last 6 months of the year 
+
+# In[72]:
+
+
+#q1.2 unpaired t test
+ls=[]
+for i in range(len(df)):
+    
+    if df.dteday[i].month <=3: 
+        ls.append(df['temp'][i])
+
+ls2=[]
+for i in range(len(df)):
+    
+    if df.dteday[i].month in [7,8,9]:  
+        ls2.append(df['temp'][i])
+
```

Comment:
avoid writing for loops on data frames . this can be very very slow . learn to conditional filter data